doc/mgmt/SONiC-Management-Framework-Image-mgmt-HLD.md

```diff
@@ -0,0 +1,225 @@
+# Image Management
+Software upgrade and image installation support for SONiC
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev  |    Date    |      Author       | Change Description |
+| :--: | :--------: | :---------------: | ------------------ |
+| 0.1  | 09/13/2019 | Arunsundar Kannan | Initial version    |
+
+# About this Manual
+This document provides general information about the Image management and installation feature implementation in SONiC.
+# Scope
+Covers Northbound interface for the Image management feature, as well as Unit Test cases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**         | **Meaning**                    |
+| ---------------- | ------------------------------ |
+| Image Management | Image installation, validation |
+
+# 1 Feature Overview
+
+Provide management framework capabilities to handle:
+
+- Image Installation
+- List available images
+- Removal of an image
+- Set default image for next boot
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide management framework support to existing SONiC capabilities with respect to Image management
+
+### 1.1.2 Configuration and Management Requirements
+- CLI style configuration and show commands
+- REST API support
+- gNMI Support
+
+### 1.1.3 Scalability Requirements
+N/A
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+The feature will use the traditional translib/transformer approach with an addition of a service running in the host which does all the "actual work". The service will use the existing sonic installer script in the host.
+
+### 1.2.2 Container
+Management Container
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+**image install**
+
+This command is used to install a new image on the alternate image partition. This command takes a path to an installable SONiC image or URL and installs the image.
+
+**show image-list**
+
+This command displays information about currently installed images. It displays a list of installed images, currently running image and image set to be loaded in next reboot.
```

Comment:
How many installed images can there be in total ? Would it be possible to mention this here ?
Based on the description on the previous command ("image install") there seem to be only two... But here we talk about a list...

Reply:
Sonic does not enforce a hard limit on the number of images that can be installed. I think space can be the only limiting factor here.  But if you want give a guideline value for the number of images that can be installed, I can add that here as well.

Reply:
As per https://github.com/Azure/SONiC/wiki/SONiC-to-SONiC-update, SONiC supports two images at a time. When you install a newer image when there are two images, the oldest one is replaced.